doc/reference/reference_sql/sql_user_guide.rst

```diff
@@ -921,18 +946,18 @@ Assignments and operations involving NULL cause NULL or UNKNOWN results. |br|
 For arithmetic, convert to the data type which can contain both operands and the result. |br|
 For explicit casts, if a meaningful result is possible, the operation is allowed. |br|
 For implicit casts, if a meaningful result is possible and the data types on both sides
-are either STRINGs or numbers (that is, are STRING or INTEGER or UNSIGNED or DOUBLE or NUMBER),
+are either STRINGs or most numeric types (that is, are STRING or INTEGER or UNSIGNED or DOUBLE but not NUMBER),
 the operation is sometimes allowed.
 
 The specific situations in this chart follow the general rules:
 
 .. code-block:: none
 
-    ~                To BOOLEAN | To number  | To STRING | To VARBINARY | To UUID
+    ~                To BOOLEAN | To numeric | To STRING | To VARBINARY | To UUID
     ---------------  ----------   ----------   ---------   ------------   -------
-    From BOOLEAN   | AAA        | S--        | A--       | ---          | ---
-    From number    | A--        | SSA        | A-A       | ---          | ---
-    From STRING    | S--        | S-S        | AAA       | A--          | SS-
+    From BOOLEAN   | AAA        | ---        | A--       | ---          | ---
+    From numeric   | ---        | SSA        | A--       | ---          | --
+    From STRING    | S--        | S--        | AAA       | A--          | SS-
     From VARBINARY | ---        | ---        | A--       | AAA          | SS-
     From UUID      | ---        | ---        | AA-       | AA-          | AAA
```

Comment:
Not sure if it should be mentioned here, but UUID cannot be implicitly cast to anything except UUID. The same for STRING/VARBINARY to UUID.

Reply:
Currently typeof(quote(uuid())) is 'string' but I expect that will change.
Currently least(uuid(),'',x'00') is '' but I won't consider that to be implicit cast for comparison.
I thought that I had tested for some other implicit casts, but now I see that they don't work.
So, okay, I'm changing to S--.